### PR TITLE
native,testlapack: some small cleanup in the eigensolver

### PR DESCRIPTION
**Pull request by vladimir-ch**
This has been on my todo list since I finished Dgeev. It is not essential but at least I can check it as done. PTAL

### CODE REVIEW

File: native/dlaqr04.go [hunk 1]
```diff
@@ -419,9 +419,6 @@ func (impl Implementation) Dlaqr04(wantt, wantz bool, n, ilo, ihi int, h []float
 			// Shuffle shifts into pairs of real shifts and pairs of
 			// complex conjugate shifts using the fact that complex
 			// conjugate shifts are already adjacent to one another.
-			// TODO(vladimir-ch): Dlaqr5 also does this shuffling.
```
Comment (vladimir-ch) [inline native/dlaqr04.go]: In fact, the shuffling here could _probably_ be removed but I'm not sure right now and it's safer to leave it.

File: native/dlaqr5.go [hunk 2]
```diff
@@ -124,9 +124,8 @@ func (impl Implementation) Dlaqr5(wantt, wantz bool, kacc22 int, n, ktop, kbot,
 	}
 
 	// Note: lapack says that nshfts must be even but allows it to be odd
-	// anyway. At the moment, we panic above if nshfts is not even, so
-	// reducing it by one is unnecessary (it seems that Dlaqr0 and Dlaqr4
-	// indeed use only even nshfts).
+	// anyway. We panic above if nshfts is not even, so reducing it by one
```
Comment (vladimir-ch) [inline native/dlaqr5.go]: We could remove the `nshfts` parameter, in Go it is not really needed.

Comment (kortschak) [inline native/dlaqr5.go]: How would that work? It seems pretty integral from a brief reading here.

Comment (vladimir-ch) [inline native/dlaqr5.go]: We could do `nshfts := len(sr)`. But but this is a very minor point not worth changing.

File: native/dlaqr04.go (hunk 1 above)
Comment (btracey) [inline native/dlaqr04.go]: Maybe change the TODO to say that instead?

**Comment by btracey** [comment]
This LGTM

File: native/dlaqr04.go (hunk 1 above)
Comment (vladimir-ch) [inline native/dlaqr04.go]: Done.

**Pull request merged**